### DISCUSSION TITLE
Implement Surge structural isomer-finding algorithm

### DISCUSSION THREAD

**Discussion by bertiewooster** [rdkit/rdkit#7101] [Ideas] [answered]
I recently learned about [Surge, an open-source chemical graph generator](https://jcheminf.biomedcentral.com/articles/10.1186/s13321-022-00604-9) aka structural isomer finder, and wondered if it's been considered for incorporation into the RDKit. Its performance seems impressive: according to [Surge's GitHub page](https://github.com/StructureGenerator/surge), it can "generate the 452458 isomers of C10H16O in 0.1s on some vanilla flavor year-2021 PC" and seems [~50x faster than MOLGEN](https://jcheminf.biomedcentral.com/articles/10.1186/s13321-022-00604-9/tables/1). It has the Apache License Version 2.0 and is implemented in C. Perhaps this could be a GSOC project.

I couldn't find any structural isomer-generation algorithm in the RDKit; is there one?

BTW I learned about Surge due to its [incorporation into WolframChemistry](https://resources.wolframcloud.com/PacletRepository/resources/WolframChemistry/IsomerGeneration/).

**Comment by greglandrum** [comment, marked as answer]
I don't think something like this is a great fit for the core RDKit, but it would be cool to have some examples of using one of these libraries together with the RDKit to do something interesting.

BTW: here's another one, from our group: https://jcheminf.biomedcentral.com/articles/10.1186/s13321-022-00677-6

**Reply by bertiewooster** [reply]
Thanks. Good to know about your group's enu, which I understand enumerates stereoisomers as well as constitutional isomers, and is part of [CombiFF](https://github.com/csms-ethz/CombiFF) on GitHub. 

Has your group considered putting CombiFF on PyPI to simplify adoption, e.g. including pre-built wheels for chips including Apple Silicon (ARM chips M1, M2, and M3)?
